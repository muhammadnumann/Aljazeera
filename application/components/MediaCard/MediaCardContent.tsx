import { MouseEventHandler } from 'react'

interface InputProps {
  onClick?: MouseEventHandler | undefined
  content: string
  slug: string
  truncateContentLength: number
  className: string
}

const MediaCardContent = (props: InputProps) => {
  return (
    <div onClick={props.onClick} className={props.className}>
      <span className="text-gray-800 dark:text-grayDark-800 mb-1 block">
        {props.slug}
      </span>
      <span className="text-gray-700 dark:text-grayDark-700">
        {props.content.substring(0, props.truncateContentLength)}
      </span>
    </div>
  )
}

MediaCardContent.defaultProps = {
  truncateContentLength: 30,
}

export default MediaCardContent
