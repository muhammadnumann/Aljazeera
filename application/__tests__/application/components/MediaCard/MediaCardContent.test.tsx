import React from 'react'
import MediaCardContent from '../../../../components/MediaCard/MediaCardContent'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe('<MediaCardContent />', () => {

    it('should render correctly', () => {
        const component = shallow(
            <MediaCardContent content="Some cool content" slug='Slug' className="style" truncateContentLength={20} onClick={void (0)} />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should apply the CSS class to the component', () => {
        const component = shallow(
            <MediaCardContent content="Some cool content" slug='Slug' className="style" />);

        expect(component.contains(
            <div className="style">
                <span className="text-gray-800 dark:text-grayDark-800 mb-1 block">
                    Slug
                </span>
                <span className="text-gray-700 dark:text-grayDark-700">
                    Some cool content
                </span>
            </div>)
        ).toEqual(true);
    });

    it('should truncate the content when "truncateContentLength" property is specified', () => {
        const component = shallow(
            <MediaCardContent content="Some cool contentSlug" slug='Slug' className="style" truncateContentLength={4} />);

        expect(component.text()).toBe('SlugSome');
    });

    it('should trigger on click event when "onClick" property is specified', () => {
        const onButtonClick = jest.fn();
        const component = shallow(
            <MediaCardContent content="Some cool contentSlug" slug='Slug' className="style" onClick={onButtonClick} />);

        component.find('div').simulate('click');
        expect(onButtonClick).toBeCalledTimes(1);
    });
});
