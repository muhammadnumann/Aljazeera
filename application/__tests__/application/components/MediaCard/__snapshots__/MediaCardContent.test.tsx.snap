// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaCardContent /> should render correctly 1`] = `
<div
  className="style"
>
  <span
    className="text-gray-800 dark:text-grayDark-800 mb-1 block"
  >
    Slug
  </span>
  <span
    className="text-gray-700 dark:text-grayDark-700"
  >
    Some cool content
  </span>
</div>
`;
